catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/UpdateOperations.java

```diff
@@ -255,6 +266,50 @@ public UpdateResponse update(UpdateStorageRequest streamUpdateRequest)
     //
     // Private helper methods
     //
+
+    private UpdateRequest rewriteRequestToAvoidHistoryConflicts(UpdateRequest updateRequest,
+            QueryResponse response) {
+        final String attributeName = updateRequest.getAttributeName();
+        if (Metacard.ID.equals(attributeName)) {
+            return updateRequest;
+        }
+
+        List<Map.Entry<Serializable, Metacard>> updatedList = response.getResults()
+                .stream()
+                .map(Result::getMetacard)
+                .map(this::toEntryById)
+                .collect(Collectors.toList());
+
+        return new UpdateRequestImpl(updatedList,
+                Metacard.ID,
+                updateRequest.getProperties(),
+                updateRequest.getStoreIds());
+    }
+
+    private boolean notFoundValidMetacards(UpdateRequest updateRequest, QueryResponse response) {
+        Set<String> originalKeys = updateRequest.getUpdates()
+                .stream()
+                .map(Map.Entry::getKey)
+                .map(Object::toString)
+                .collect(Collectors.toSet());
+        Set<String> responseKeys = response.getResults()
+                .stream()
+                .map(Result::getMetacard)
+                .map(m -> m.getAttribute(updateRequest.getAttributeName()))
+                .map(Attribute::getValue)
+                .map(Object::toString)
+                .collect(Collectors.toSet());
+        boolean notSameSize = response.getResults()
+                .size() != updateRequest.getUpdates()
+                .size();
+        return !Sets.difference(originalKeys, responseKeys)
```

Comment:
Can we remove `notSameSize`.  I think the set difference should always give the correct answer and short circuit before ORing `notSameSize`.